projects/plugins/protect/src/js/hooks/use-protect-data/index.js

```diff
@@ -74,6 +84,16 @@ export default function useProtectData() {
 		} )
 	);
 
+	const { refreshStatus } = useDispatch( STORE_ID );
+
+	useEffect( () => {
+		if ( true !== statusIsFetching && isRegistered && ! status.status ) {
+			refreshStatus();
+		}
+		// We don't want to run the effect if status changes. Only on changes on isRegistered.
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ isRegistered ] );
```

Comment:
IMO we should move out of `useEffect` to listen to state updates, we already have it by `subscribe`: https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#subscribe 

We could get our specific store `subscribe` (to not listen for all registered store) from `store.instantiate`:
https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#registering-a-store

Prefer to use it and follow well the [`flux`](https://facebook.github.io/flux/docs/in-depth-overview) pattern.